### ISSUE TITLE
pyyaml not in requirements.txt

### ISSUE THREAD

**Issue by DerPhysikeR** [ljvmiranda921/pyswarms#300]
Hi everybody, thank you for pyswarms!

Is there a reason why pyyaml is only defined as a dev dependency?

**Describe the bug**
Pyswarms can not be imported after installation, since it depends on pyyaml in pyswarms/utils/reporter/reporter.py, which is not defined as a dependency.

**To Reproduce**
 1. install pyswarms with pipenv
 2. try to import pyswarms
The error message shown above should appear.

**Expected behavior**
pyswarms should be imported.

**Environment (please complete the following information):**
 - OS: Linux
 - Version: Manjaro (most recent version 2019-02-20)
 - PySwarms Version: 1.0.2
 - Python Version: 3.7

**Additional context**

This is the error message I get, when I try to import pyswarms:

```python
Python 3.7.2 (default, Jan 10 2019, 23:51:51) 
[GCC 8.2.1 20181127] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyswarms
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/paul/.local/share/virtualenvs/netfitter-x944HxYD/lib/python3.7/site-packages/pyswarms/__init__.py", line 17, in <module>
    from .single import global_best, local_best, general_optimizer
  File "/home/paul/.local/share/virtualenvs/netfitter-x944HxYD/lib/python3.7/site-packages/pyswarms/single/__init__.py", line 17, in <module>
    from .global_best import GlobalBestPSO
  File "/home/paul/.local/share/virtualenvs/netfitter-x944HxYD/lib/python3.7/site-packages/pyswarms/single/global_best.py", line 65, in <module>
    from ..backend.operators import compute_pbest
  File "/home/paul/.local/share/virtualenvs/netfitter-x944HxYD/lib/python3.7/site-packages/pyswarms/backend/__init__.py", line 7, in <module>
    from .generators import *
  File "/home/paul/.local/share/virtualenvs/netfitter-x944HxYD/lib/python3.7/site-packages/pyswarms/backend/generators.py", line 18, in <module>
    from ..utils.reporter import Reporter
  File "/home/paul/.local/share/virtualenvs/netfitter-x944HxYD/lib/python3.7/site-packages/pyswarms/utils/__init__.py", line 3, in <module>
    from .reporter.reporter import Reporter
  File "/home/paul/.local/share/virtualenvs/netfitter-x944HxYD/lib/python3.7/site-packages/pyswarms/utils/reporter/__init__.py", line 3, in <module>
    from .reporter import Reporter
  File "/home/paul/.local/share/virtualenvs/netfitter-x944HxYD/lib/python3.7/site-packages/pyswarms/utils/reporter/reporter.py", line 9, in <module>
    import yaml
ModuleNotFoundError: No module named 'yaml'
```
It can easily be fixed by installing pyyaml by hand, but shouldn't it be in the requirements if it is required?

**Comment by ljvmiranda921** [comment]
Hi, I think this is a mistake on our part. Pyyaml should be in `requirements.txt`
Would you like to open up a Pull Request for that?

**Comment by DerPhysikeR** [comment]
Yes, I will do that.

**Comment by ljvmiranda921** [comment]
Awesome!! Don't hesitate to ping us if you have any questions!

**Comment by DerPhysikeR** [comment]
I'm sorry, but I will not be able to look into this issue for the next few days. If somebody else wants to make a pull request, feel free to do so.

**Comment by citomcclure** [comment]
Hi @ljvmiranda921 , I'd be happy to take this one, but we can wait a little bit to see if a first-timer would like to take it. Thank you!

**Comment by ctangell** [comment]
I'd like to try to take this on, if that's alright.

**Comment by ljvmiranda921** [comment]
Hi @citomcclure sorry for the late reply, yes you can take this!
Hello @ctangell , sorry but @citomcclure made dibs on this 5 days ago. But you can still contribute! I can refer you to Issues #297 or #264 if you're interested!

**Comment by stevenbw** [comment]
Closed by #316
